src/overlays/actors/ovl_Demo_Du/z_demo_du.c

```diff
@@ -1,228 +1,1083 @@
+        Item_Give(globalCtx, ITEM_MEDALLION_FIRE);
+
+        rotY = this->actor.world.rot.y + 0x8000;
+        player->actor.shape.rot.y = rotY;
+        player->actor.world.rot.y = rotY;
+    }
+}
+
+void DemoDu_CsFireMedallion_AdvanceTo02(DemoDu* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        npcAction = globalCtx->csCtx.npcActions[2];
+        if (npcAction != NULL && npcAction->action != 1) {
+            this->updateIndex = CS_FIREMEDALLION_SUBSCENE(2);
+            this->drawIndex = 1;
+            DemoDu_CsFireMedallion_SpawnDoorWarp(this, globalCtx);
+        }
+    }
+}
+
+void DemoDu_CsFireMedallion_AdvanceTo03(DemoDu* this) {
+    if (this->actor.shape.yOffset >= 0.0f) {
+        this->updateIndex = CS_FIREMEDALLION_SUBSCENE(3);
+        this->actor.shape.yOffset = 0.0f;
+    }
+}
+
+void DemoDu_CsFireMedallion_AdvanceTo04(DemoDu* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        npcAction = globalCtx->csCtx.npcActions[2];
+        if (npcAction != NULL && npcAction->action != 2) {
+            Animation_Change(&this->skelAnime, &D_06000800, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000800), 2, 0.0f);
+            this->updateIndex = CS_FIREMEDALLION_SUBSCENE(4);
+        }
+    }
+}
+
+void DemoDu_CsFireMedallion_AdvanceTo05(DemoDu* this, s32 didAnimFinished) {
+    if (didAnimFinished) {
+        Animation_Change(&this->skelAnime, &D_06000D00, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000D00), 0, 0.0f);
+        this->updateIndex = CS_FIREMEDALLION_SUBSCENE(5);
+    }
+}
+
+void DemoDu_CsFireMedallion_AdvanceTo06(DemoDu* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        npcAction = globalCtx->csCtx.npcActions[6];
+        if (npcAction != NULL && npcAction->action == 2) {
+            this->updateIndex = CS_FIREMEDALLION_SUBSCENE(6);
+            func_80969F38(this, globalCtx);
+        }
+    }
+}
+
+void DemoDu_UpdateCs_FM_00(DemoDu* this, GlobalContext* globalCtx) {
+    DemoDu_CsFireMedallion_AdvanceTo01(this, globalCtx);
+}
+
+void DemoDu_UpdateCs_FM_01(DemoDu* this, GlobalContext* globalCtx) {
+    DemoDu_CsFireMedallion_AdvanceTo02(this, globalCtx);
+}
+
+void DemoDu_UpdateCs_FM_02(DemoDu* this, GlobalContext* globalCtx) {
+    func_80969FB4(this, globalCtx);
+    DemoDu_UpdateSkelAnime(this);
+    DemoDu_CsFireMedallion_AdvanceTo03(this);
+}
+
+void DemoDu_UpdateCs_FM_03(DemoDu* this, GlobalContext* globalCtx) {
+    DemoDu_UpdateBgCheckInfo(this, globalCtx);
+    DemoDu_UpdateSkelAnime(this);
+    DemoDu_CsFireMedallion_AdvanceTo04(this, globalCtx);
+}
+
+void DemoDu_UpdateCs_FM_04(DemoDu* this, GlobalContext* globalCtx) {
+    s32 didAnimFinished;
+
+    DemoDu_UpdateBgCheckInfo(this, globalCtx);
+    didAnimFinished = DemoDu_UpdateSkelAnime(this);
+    DemoDu_CsFireMedallion_AdvanceTo05(this, didAnimFinished);
```

Comment:
```suggestion
    DemoDu_CsFireMedallion_AdvanceTo05(this, DemoDu_UpdateSkelAnime(this));
```
Hypothetically it could be written like this. Not sure we want to.